### DISCUSSION TITLE
Feature Request: support nanosecond in interval type

### DISCUSSION THREAD

**Discussion by axhf** [duckdb/duckdb#15415] [Ideas]
`nanosecond` is supported for read/write: #12440.

It seems not fully supported yet. For example, `interval`:
* From the document: [interval](https://duckdb.org/docs/sql/data_types/interval.html), [datepart functions](https://duckdb.org/docs/sql/functions/datepart.html)
* Or from the code: [interval_t](https://github.com/duckdb/duckdb/blob/main/src/include/duckdb/common/types/interval.hpp#L25C1-L27C17)

Similar requests: https://github.com/duckdb/duckdb/discussions/12020#discussioncomment-9415433

**Comment by MarcoGorelli** [comment]
Seems there's no way to use `interval` on a `timestamp_ns` column currently without losing nanosecond info:
```
In [42]: duckdb.sql("""from (values (timestamp_ns '2020-01-01T01:02:03.123456789')) df(d) select d, d + interval 1 hour as b""")
Out[42]:
┌───────────────────────────────┬────────────────────────────┐
│               d               │             b              │
│         timestamp_ns          │         timestamp          │
├───────────────────────────────┼────────────────────────────┤
│ 2020-01-01 01:02:03.123456789 │ 2020-01-01 02:02:03.123456 │
└───────────────────────────────┴────────────────────────────┘
```

**Comment by d-chambers** [comment]
I am also running into this. 

Related, but timedelta64[ns] don't round drop to arrays or dataframes like datetime64[ns], but interestingly their dtype is preserved. 

```python
import duckdb
import numpy as np

# Test datetime 
dt_ar = np.array([11], dtype='datetime64[ns]') # array(['1970-01-01T00:00:00.000000011'], dtype='datetime64[ns]')
dt_ar_out = duckdb.sql("SELECT * from dt_ar").fetchnumpy()['column0']
assert np.all (dt_ar == dt_ar_out)  # works

td_ar = np.array([11], dtype='timedelta64[ns]') # array([11], dtype='timedelta64[ns]')
td_ar_out = duckdb.sql("SELECT * from td_ar").fetchnumpy()['column0']
assert np.all (td_ar == td_ar_out)  # fails as td_ar_out is array([0], dtype='timedelta64[ns]')
```
